plasma_framework/docs/design/ald-exit-game-definitions-and-properties.md

```diff
@@ -0,0 +1,134 @@
+# ALD Exit Game Definitions And Properties
+
+## New definitions from ALD (abstract layer design)
+Based on [MoreVP document](https://github.com/omisego/elixir-omg/blob/master/docs/morevp.md). This section adds some new definition introduced from ALD.
+
+### SpendingCondition Fulfilled
+1. Checks an input tx and the spending tx, whether the spending tx fulfil the requirement of spending the output of input tx or not.
+1. Basic checks:
+    - Spending tx input points to the output of input tx.
+    - Spending tx fulfil the “lock” condition of the output
+1. This checks on (at most) tx level, does not check things on Plasma protocol level.
+
+### Tx Standard Finalization
+1. MVP:
+    - Tx is in a plasma block. (Checks inclusion proof of plasma block)
+    - It is confirmed by user. (Checks confirm signature is valid for the inclusion block)
+1. MoreVP:
+    - Tx is in a plasma block. (Checks inclusion proof of plasma block)
+
+### Tx In-flight Finalization
+Only with MoreVP, and only need the existence of an tx (Nothing specific is checked)
+
+### Tx Protocol Finalized:
+1. MoreVp tx: needs to be in-flight finalized
+1. MVP tx: needs to be standard finalized.
+
+## Exit Properties
+
+### General Properties
+
+#### Valid Spending Tx:
+1. Needs to be protocol finalized
+1. Fulfils the spending condition of the output
+
+#### Input/Output finalized
+1. The input/output has been withdrawn/processed.
+1. Or The output has been used as an input of a processed IFE.
+
+#### Input/Output withdrawable
+The input/output is not finalized already
+
+### In-flight Exit
+
+#### InFlight exit to be exitable
+1. Inputs of the exiting tx should be Standard Finalized (Protocol finalized is not enough)
+1. The state transition of the tx is valid
+1. All inputs fulfill the spending condition with the exiting tx
+1. The transaction itself is Protocol finalized
+
+#### Competing tx
+1. Needs to be protocol finalized
+1. Fulfils the spending condition of an input of the InFlight exiting tx
+
+#### Canonical tx
+1. The tx is exitable
+1. No competing tx that has higher or same priority (txPos) then the exiting one. (PS, same priority is for the case both txs are in-flight)
+1. All inputs of the tx are withdrawable
```

Comment:
ah...I think the purpose is not to point to the "flag" mechanism that we are using. I just want to say a tx is not canonical if input is already exited.

Change the sentence to be using "finalized" instead, which is defined in above the withdrawal section and is a better word choice here:
"1. All inputs of the tx are not finalized"

Reply:
hmm well, this Is indeed a bit extending the definition of "canonicity".

Original definition in spec is: "A transaction is “canonical” if none of its inputs were previously spent in any other transaction..."

I extend it to: "...were spent in any other transaction or exit...", does this make sense? Or shall we leave this to other place of definition on finality?

Reply:
The problem we are dealing here is a security hole in original MoreVP construction. What you are doing is fixing the construction. That construction is already quite hard to process and by doing that I think you are making it a bit harder by expanding canonicity notion. The notion by itself is still very useful since it allows to solve the conflicts between normal users in situation of operator going down. The bug itself does not renders canonicity useless.

What I'm proposing is - don't try to fix it by redefining. Just point out that there is the problem and there is a solution.

Reply:
I see. Agree with that so I removing the extension of canonicity and add a line noting the current situation.